### PR TITLE
docs: replace the deprecated TuiFieldError with TuiError and TuiFieldErrorPipe in examples

### PR DESCRIPTION
**Pull request by ddubrava**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [x] Documentation content changes

## What is the current behavior?

Examples use the deprecated TuiFieldError instead of a TuiFieldErrorPipe.
TuiError has a link to the TuiFieldError instead of the TuiFieldErrorPipe.

## What is the new behavior?

All examples are updated to use the TuiFieldErrorPipe.
TuiError has a link to the TuiFieldErrorPipe.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by ddubrava** [comment]
Hey, guys! There is a problem that examples with a TuiFieldErrorPipe do not work on stackblitz.
There is the error that 'no value accessor for form control with name....'.
I faced with the same problem in jest + TuiFieldErroPipe, so there are already some discussions about the missing NgControl in our messenger.

**Review by splincode — APPROVED** [review]

**Comment by waterplea** [comment]
> Hey, guys! There is a problem that examples with a TuiFieldErrorPipe do not work on stackblitz. There is the error that 'no value accessor for form control with name....'. I faced with the same problem in jest + TuiFieldErroPipe, so there are already some discussions about the missing NgControl in our messenger.

Unfortunately `ngcc` is broken on StackBlitz with Taiga UI. When 3.0 is out with Ivy distribution it will go away.

**Review by waterplea — APPROVED** [review]

File: projects/demo/src/modules/components/text-area/examples/4/index.html [hunk 1]
```diff
@@ -35,6 +35,9 @@
         >
             Type it
         </tui-text-area>
-        <tui-field-error formControlName="testValue1"></tui-field-error>
+        <tui-error
+            formControlName="testValue1"
+            [error]="[] | tuiFieldError | async"
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/text-area/examples/4/index.html:40]: It's curious why the behaviour was changed 🤔

Comment (nsbarsukov) [reply projects/demo/src/modules/components/text-area/examples/4/index.html:40]: anyway, it not a problem of current PR.
I will create an issue for this problem.

**Pull request merged**